File: Time Needed to Inform All Employees/Time Needed to Inform All Employees.cpp
class Solution
{
public:
    int dfs(int node, int par, vector<vector<int>> &adj, vector<int> &informTime)
    {
        int ret = 0;
        for (auto it : adj[node])
        {
            if (it != par)
            {
                ret = max(ret, dfs(it, node, adj, informTime));
            }
        }
        return ret + informTime[node];
    }
    int numOfMinutes(int n, int headID, vector<int> &manager, vector<int> &informTime)
    {
        vector<vector<int>> adj(n);
        for (int i = 0; i < n; i++)
        {
            if (~manager[i])
                adj[manager[i]].push_back(i);
        }
        return dfs(headID, -1, adj, informTime);
    }
};

File: Time Needed to Inform All Employees/Time Needed to Inform All Employees.py
class Solution:
    def numOfMinutes(self, n: int, headID: int, manager: List[int], informTime: List[int]) -> int:
        adj = [[] for _ in range(n)]
        for i in range(n):
            if manager[i]!=-1:
                adj[manager[i]].append(i)
        def dfs(node:int,par:int)->int:
            ret=0
            for x in adj[node]:
                if x != par:
                    ret=max(ret,dfs(x,node))
            return ret+informTime[node]
        return dfs(headID,-1)